planner: TestGlobalStatsAndSQLBindingWithConcurrency can be flaky

## Enhancement

TestGlobalStatsAndSQLBindingWithConcurrency test can be flaky if auto_analyze executes before the manual analyze.

--- FAIL: TestGlobalStatsAndSQLBindingWithConcurrency (0.51s)
    testkit.go:128: RefreshSession rand seed: 1750451259174826000
    testkit.go:320: 
        	Error Trace:	/Users/terrypurcell/tidb/pkg/testkit/testkit.go:320
        	            				/Users/terrypurcell/tidb/pkg/statistics/handle/globalstats/global_stats_internal_test.go:452
        	            				/Users/terrypurcell/tidb/pkg/statistics/handle/globalstats/global_stats_test.go:979
        	Error:      	Should be true
        	Test:       	TestGlobalStatsAndSQLBindingWithConcurrency
        	Messages:   	select * from trange where a<100 doesn't have plan TableFullScan, full plan [[IndexLookUp_10 9.50 root partition:p0 ] [├─IndexRangeScan_8(Build) 9.50 cop[tikv] table:trange, index:a(a) range:[-inf,100), keep order:false] [└─TableRowIDScan_9(Probe) 9.50 cop[tikv] table:trange keep order:false]]
FAIL
exit status 1

--- a/pkg/statistics/handle/globalstats/global_stats_internal_test.go
+++ b/pkg/statistics/handle/globalstats/global_stats_internal_test.go
@@ -439,6 +439,8 @@ func testGlobalStatsAndSQLBinding(tk *testkit.TestKit) {
 	tk.MustExec("insert into trange values " + strings.Join(vals, ","))
 	tk.MustExec("insert into tlist values " + strings.Join(listVals, ","))
 
+	// Disable auto analyze to ensure that stats are not automatically collected
+	tk.MustExec("set @@global.tidb_enable_auto_analyze='OFF'")
 	// before analyzing, the planner will choose TableScan to access the 1% of records
 	tk.MustHavePlan("select * from thash where a<100", "TableFullScan")
 	tk.MustHavePlan("select * from trange where a<100", "TableFullScan")
@@ -470,4 +472,6 @@ func testGlobalStatsAndSQLBinding(tk *testkit.TestKit) {
 	tk.MustHavePlan("select * from thash where a<100", "TableFullScan")
 	tk.MustHavePlan("select * from trange where a<100", "TableFullScan")
 	tk.MustHavePlan("select * from tlist where a<1", "TableFullScan")
+	// Reset auto analyze after test
+	tk.MustExec("set @@global.tidb_enable_auto_analyze='ON'")
 }

